### PR TITLE
GH-38239: [CI][Python] Disable -W error on Python CI jobs temporarily

### LINKED ISSUE #38239: [CI][Python] test-conda-python is failing due to new DeprecationWarning raised on aiohttp
### Describe the bug, including details regarding any error messages, version, and platform.

On the test-conda-python nightly builds we have enabled Warnings as errors: https://github.com/apache/arrow/pull/37018
This is currently causing some nightly failures due to a new DeprecationWarning being raised by aiohttp due to s3fs:

https://github.com/ursacomputing/crossbow/actions/runs/6489587884/job/17624099591#step:6:8770
```
self = <aiohttp.client.ClientSession object at 0x7f184812d770>
name = '_resolve_charset'
val = <function _default_fallback_charset_resolver at 0x7f184b380260>

    def __setattr__(self, name: str, val: Any) -> None:
        if name not in self.ATTRS:
>           warnings.warn(
                "Setting custom ClientSession.{} attribute "
                "is discouraged".format(name),
                DeprecationWarning,
                stacklevel=2,
            )
E           DeprecationWarning: Setting custom ClientSession._resolve_charset attribute is discouraged

opt/conda/envs/arrow/lib/python3.10/site-packages/aiohttp/client.py:354: DeprecationWarning
```

[test-conda-python-3.10](https://github.com/ursacomputing/crossbow/actions/runs/6489587884/job/17624099591)
[test-conda-python-3.11](https://github.com/ursacomputing/crossbow/actions/runs/6489588138/job/17624100425)
[test-conda-python-3.12](https://github.com/ursacomputing/crossbow/actions/runs/6489588226/job/17624100615)
[test-conda-python-3.8](https://github.com/ursacomputing/crossbow/actions/runs/6489588248/job/17624100846)
[test-conda-python-3.9](https://github.com/ursacomputing/crossbow/actions/runs/6489588349/job/17624101074)

### Component(s)

Continuous Integration, Python

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
* Closes: #38239

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit test-conda-python-3.11

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit test-conda-python-3.11

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit test-conda-python-3.11

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit test-conda-python-3.11

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit test-conda-python-3.11

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit test-conda-python-3.11

**Comment by danepitkin** [comment]
It would be nice to disable all `DeprecationWarning`s and `FutureWarning`s since they pop up a lot in our dependencies 😬 I hadn't anticipated that when enabling initially.

File: dev/tasks/tasks.yml [hunk 1]
```diff
@@ -1241,7 +1241,7 @@ tasks:
     params:
       env:
         PYTHON: "{{ python_version }}"
-        PYTEST_ARGS: "-W error"
+        PYTEST_ARGS: "-W error -W \"ignore:Setting custom ClientSession:DeprecationWarning\""
```
[the remark this answers is not in the document]
Comment (assignUser) [reply dev/tasks/tasks.yml:1244]: The syntax you used works when used directly in a gha https://github.com/assignUser/test-repo-a/actions/runs/6502900081/job/17662621207

It's the jinja processing that breaks it as it strips the \.
```suggestion
        PYTEST_ARGS: '-W error -W \"ignore:Setting custom ClientSession:DeprecationWarning\"'
```

**Comment by assignUser** [comment]
@github-actions crossbow submit test-conda-python-3.11

**Comment by assignUser** [comment]
:/
```
+ pytest -r s -W error -W '"ignore:Setting' custom 'ClientSession:DeprecationWarning"' --pyargs pyarrow
ERROR: while parsing the following warning configuration:

  "ignore:Setting

This error occurred:

invalid action: '"ignore
```
The envvar seems to be set correctly? https://github.com/ursacomputing/crossbow/actions/runs/6502979236/job/17662809445#step:6:1677

**Comment by assignUser** [comment]
The actual problem is seemingly not the syntax/string but passing the flags as one envvar?
```
 WARNS="-Werror -Werror"
python $WARNS
Invalid -W option ignored: invalid action: 'error -Werror
``` 
Using the flags directly does not cause an error...

**Comment by assignUser** [comment]
To get around the fact that python doesn't seems to detect to flags but rather one malformed flag when passed via envvar I manually edited the workflow to add [`PYTHONWARNINGS`](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONWARNINGS) to the docker container with `error,ignore:...` and that seems to work but there are also other deprecation warnings I guess? https://github.com/ursacomputing/crossbow/actions/runs/6503439585/job/17663911614#step:6:3043

**Comment by raulcd** [comment]
@github-actions crossbow submit test-conda-python-3.11

**Comment by jorisvandenbossche** [comment]
@raulcd I am afraid that will fail, that was the first thing I tried as well .. https://github.com/apache/arrow/pull/38238/commits/78460fa8522cd5a02d4f1401e9fca30f59b89421

**Comment by jorisvandenbossche** [comment]
And also locally that fails:

``` 
$ export PYTEST_ARGS="-W error -W 'ignore:Setting custom ClientSession:DeprecationWarning'"
$ PYTHONASYNCIODEBUG=1 pytest python/pyarrow/tests/test_dataset.py::test_open_dataset_from_uri_s3_fsspec $PYTEST_ARGS
ERROR: while parsing the following warning configuration:

  'ignore:Setting

This error occurred:

invalid action: "'ignore"
```

So indeed it seems to be something with passing it as an env var as Jakob mentions, and how spaces/quotes get treated in that case ..

**Comment by raulcd** [comment]
This works:
```
pytest -r s $(PYTEST_ARGS)
```

**Comment by jorisvandenbossche** [comment]
That gives me `PYTEST_ARGS: command not found`, but that might depend on the bash terminal I am using?

(the tests then do run, so that might mask the error)

**Comment by raulcd** [comment]
@github-actions crossbow submit test-conda-python-3.11

**Comment by raulcd** [comment]
It is green but the whole `PYTEST_ARGS` were ignored:
```
 PYTEST_ARGS: command not found
 ```

**Comment by jorisvandenbossche** [comment]
Yep, that's what I said above ;) I see that locally as well, and then the tests are green, because also the `-W error` (which is causing the error) wasn't actually passed.

**Comment by raulcd** [comment]
@github-actions crossbow submit test-conda-python-3.11

**Comment by raulcd** [comment]
@jorisvandenbossche I agree with your statement here: https://github.com/apache/arrow/pull/38259#issuecomment-1761345523
I've removed the `-W error` and created a follow up issue here: https://github.com/apache/arrow/issues/38262

**Review by raulcd — APPROVED** [review]
:+1:

**Pull request merged**